pkg/vkMachinery/forge/forge.go

```diff
@@ -112,12 +115,16 @@ func forgeVKContainers(
 
 func forgeVKPodSpec(
 	vkNamespace string,
-	homeCluster, remoteCluster *discoveryv1alpha1.ClusterIdentity, opts *VirtualKubeletOpts,
-	resourceOffer *sharingv1alpha1.ResourceOffer) v1.PodSpec {
-	nodeName := virtualKubelet.VirtualNodeName(remoteCluster)
+	homeCluster *discoveryv1alpha1.ClusterIdentity, virtualNode *virtualkubeletv1alpha1.VirtualNode, opts *VirtualKubeletOpts) v1.PodSpec {
+	var nodeName string
+	if opts.NodeName != "" {
+		nodeName = opts.NodeName
```

Comment:
Does opts.NodeName have still meaning or it can be replaced by virtualNode.Name in every case?